src/secrets.ts

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2023 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { type SafeStorage, safeStorage } from "electron";
+import Store from "electron-store";
+
+/**
+ * Save secrets on a local machine.
+ * These secrets are associated with a local machine user, not an Element user.
+ * CAUTION: They are not cleared when a user logs out of Element.
```

Comment:
Hmm, what is the reason for this? This doesn't seem desirable, we go out of our way to clear all data when you log out using `electron-clear-data` via the `clearStorage` IPC

Reply:
Of couse we can only do this on a trusted platform/filesystem, I am aware this is against the spirit of the remaining code base, thus the comment.

The feature of this PR is intended for use in https://github.com/matrix-org/matrix-react-sdk/pull/11776 (Store the ssss key on the platform so it is easier to use and harder to lose). Not clearing the key reduces friction, as it enables a user to log into Element again and have the device automatically verified.

What if we make clearing the secrets a feature flag that explicitly warn about the implications?